academy/3-ibc/6-ica.md

```diff
@@ -16,635 +15,297 @@ tags:
-        }
-    }
-}
-```
+* Authenticate Interchain account owners.
+* Track the associated Interchain account address for an owner.
+* Send packets on behalf of an owner (after authentication).
 
-A handler is needed to interpret what actions to perform based on the type url of the Any. A router could be used, or more simply a switch statement:
+<HighlightBox type="docs">
 
-```go
-func handleAny(any *codectypes.Any) error {
-    switch any.TypeURL {
-        case banktypes.MsgSend:
-            msgResponse, err := unpackBankMsgSendResponse(any)
-            if err != nil {
-                return err
-            }
-
-            handleBankSendMsg(msgResponse)
-
-        case stakingtypes.MsgDelegate:
-            msgResponse, err := unpackStakingDelegateResponse(any)
-            if err != nil {
-                return err
-            }
-
-            handleStakingDelegateMsg(msgResponse)
-
-            case transfertypes.MsgTransfer:
-            msgResponse, err := unpackIBCTransferMsgResponse(any)
-            if err != nil {
-                return err
-            }
-
-            handleIBCTransferMsg(msgResponse)
-
-        default:
-            return
-    }
-}
-```
+Originally, when ICA was first introduced in ibc-go v3, developers had to develop a custom authentication module as an IBC application that was wrapped by the ICS-27 module acting as middleware. In ibc-go v6, a refactor of ICA took place that enabled Cosmos SDK modules (`x/auth`, `x/gov`, or `x/group`) to act as generic authentication modules that required no extra development.
+<br/><br/>
+A `MsgServer` was added to the ICA controller submodule to facilitate this.
+<br/><br/>
+More information regarding the details and context for the redesign can be found in [ADR-009](https://github.com/cosmos/ibc-go/blob/main/docs/architecture/adr-009-v6-ics27-msgserver.md) or in a [dedicated blog post](https://medium.com/the-interchain-foundation/ibc-go-v6-changes-to-interchain-accounts-and-how-it-impacts-your-chain-806c185300d7) on the topic.
+<br/><br/>
+For now, the legacy API remains available for those developers who have already built custom IBC authentication modules, but it will be deprecated in the future.
 
-## Example integration
+</HighlightBox>
 
-[Here](https://github.com/cosmos/interchain-accounts-demo) is  an end-to-end working demo of interchain accounts. You will notice that it contains a similar `app.go` to the generic one which follows.
+<HighlightBox type="info">
 
+**SDK Security Model**
+<br/><br/>
+SDK modules on a chain are assumed to be trustworthy. For example, there are no checks to prevent an untrustworthy module from accessing the bank keeper.
+<br/><br/>
+The implementation of [ICS-27](https://github.com/cosmos/ibc/blob/master/spec/app/ics-027-interchain-accounts/README.md) on [ibc-go](https://github.com/cosmos/ibc-go/tree/main/modules/apps/27-interchain-accounts) uses this assumption in its security considerations. It assumes that:
```

Comment:
not exactly sure yet how to refactor but i find this section a bit confusing as to what the security implications actually are from ics27 then just reading this description

Reply:
@srdtrk Your opinion is probably sought here.

Reply:
ill check in on this!

Reply:
@srdtrk Can you prepare a suggestion?

Reply:
I don't think there is need for changes @xavierlepretre. The issue is consistent with the document